### DISCUSSION TITLE
Access Lambda context with lambda_http

### DISCUSSION THREAD

**Discussion by floric** [aws/aws-lambda-rust-runtime#436] [Q&A] [answered]
Hey :)

at first thanks for your great work and invest in the Rust ecosystem. :) Now my question:

is it possible to access the Lambda context (part of `LambdaEvent`) if we use `lambda_http::run`? I really enjoy the abstraction of this package but now I'm missing some variables like the trace id or the request id. Or is it possible to use `lambda_runtime` and `lambda_http` together to achieve this goal?

Kind regards, Florian

**Comment by nmoutschen** [comment, marked as answer]
Hey @floric !

It's definitively possible, but I do realise we might not have documented this in the best way.

The Lambda context is stored as a [Request extension](https://docs.rs/http/latest/http/request/struct.Request.html#method.extensions). You can use the [`RequestExt`](https://docs.rs/lambda_http/latest/lambda_http/ext/trait.RequestExt.html) trait, which will expose the Lambda context by doing `req.lambda_context()`.

**Reply by floric** [reply]
great, that sounds great. Traits are so helpful and deeply hidden at the same time :D I will give it a try.
